yt_dlp/extractor/reddit.py

```diff
@@ -159,6 +177,34 @@ def add_thumbnail(src):
                 'duration': int_or_none(reddit_video.get('duration')),
             }
 
+        # Check if media is hosted on reddit as an image (but also available as video):
+        reddit_video = traverse_obj(data, ('preview', 'images', 0, 'variants', 'mp4'))
+        if reddit_video:
+            display_id = video_id
+            video_id = self._search_regex(
+                r'https?://i\.redd\.it/(?P<id>[^\.]+)', video_url, 'video_id', default=display_id)
+
+            formats = [
+                {
+                    'url': unescapeHTML(x['url']),
+                    'width': int_or_none(x.get('width')),
+                    'height': int_or_none(x.get('height')),
+                    'ext': 'mp4'
+                }
+                for x in filter(
+                    lambda x: isinstance(x, dict),
+                    [reddit_video.get('source')] + traverse_obj(
+                        reddit_video, 'resolutions', expected_type=list, default=[]))
+            ]
```

Comment:
The original GIF file is missing from the format list.

Reply:
Based on an issue search I didn;t think this was in scope for yt-dlp (Skeb [doesn't](https://github.com/yt-dlp/yt-dlp/pull/1916#pullrequestreview-825552814) and YTMND was [rejected](https://github.com/yt-dlp/yt-dlp/issues/1051) outright), but having looked at the code I see that Imgur, Gfycat, Redgifs, and Gamejolt all do it. Added formats for all GIF sizes.

Reply:
Gifs are allowed because while they are technically not video, they can be treated like one